clippy_utils/src/hir_utils.rs

```diff
@@ -288,21 +290,54 @@ impl HirEqInterExpr<'_, '_, '_> {
                     && self.eq_expr(l_receiver, r_receiver)
                     && self.eq_exprs(l_args, r_args)
             },
+            (ExprKind::Path(l), ExprKind::Path(r)) => self.eq_qpath(l, r),
             (&ExprKind::Repeat(le, ll), &ExprKind::Repeat(re, rl)) => {
                 self.eq_expr(le, re) && self.eq_array_length(ll, rl)
             },
             (ExprKind::Ret(l), ExprKind::Ret(r)) => both(l, r, |l, r| self.eq_expr(l, r)),
-            (ExprKind::Path(l), ExprKind::Path(r)) => self.eq_qpath(l, r),
             (&ExprKind::Struct(l_path, lf, ref lo), &ExprKind::Struct(r_path, rf, ref ro)) => {
                 self.eq_qpath(l_path, r_path)
                     && both(lo, ro, |l, r| self.eq_expr(l, r))
                     && over(lf, rf, |l, r| self.eq_expr_field(l, r))
             },
             (&ExprKind::Tup(l_tup), &ExprKind::Tup(r_tup)) => self.eq_exprs(l_tup, r_tup),
             (&ExprKind::Unary(l_op, le), &ExprKind::Unary(r_op, re)) => l_op == r_op && self.eq_expr(le, re),
-            (&ExprKind::Array(l), &ExprKind::Array(r)) => self.eq_exprs(l, r),
-            (&ExprKind::DropTemps(le), &ExprKind::DropTemps(re)) => self.eq_expr(le, re),
-            _ => false,
+            // Else branches for branches above, grouped as per `match_same_arms`.
+            (&ExprKind::AddrOf(..)
+            | &ExprKind::Array(..)
+            | &ExprKind::Assign(..)
+            | &ExprKind::AssignOp(..)
+            | &ExprKind::Binary(..)
+            | &ExprKind::Block(..)
+            | &ExprKind::Box(..)
+            | &ExprKind::Break(..)
+            | &ExprKind::Call(..)
+            | &ExprKind::Cast(..)
+            | &ExprKind::Continue(..)
+            | &ExprKind::DropTemps(..)
+            | &ExprKind::Field(..)
+            | &ExprKind::Index(..)
+            | &ExprKind::If(..)
+            | &ExprKind::Let(..)
+            | ExprKind::Lit(..)
+            | &ExprKind::Loop(..)
+            | &ExprKind::Match(..)
+            | &ExprKind::MethodCall(..)
+            | ExprKind::Path(..)
+            | &ExprKind::Repeat(..)
+            | ExprKind::Ret(..)
```

Comment:
Some of those don't have a `&`. I would add it for consistency.

Reply:
It did feel weird. Should we add them also in the positive cases?

Reply:
I wouldn't add them there. Probably Clippy will complain if you do :D